pkg/operator/ceph/file/mds/config.go

```diff
@@ -71,7 +71,10 @@ func (c *Cluster) setDefaultFlagsMonConfigStore(mdsID string) error {
 	configOptions := make(map[string]string)
 
 	// Set mds cache memory limit to the best appropriate value
-	if !c.fs.Spec.MetadataServer.Resources.Limits.Memory().IsZero() {
+	if !c.fs.Spec.MetadataServer.Resources.Requests.Memory().IsZero() {
+		mdsCacheMemoryRequest := float64(c.fs.Spec.MetadataServer.Resources.Requests.Memory().Value()) * mdsCacheMemoryResourceFactor
+		configOptions["mds_cache_memory_limit"] = strconv.Itoa(int(mdsCacheMemoryRequest))
+	} else if !c.fs.Spec.MetadataServer.Resources.Limits.Memory().IsZero() {
```

Comment:
Let's just change the order to check for limits first, then requests

Reply:
done.